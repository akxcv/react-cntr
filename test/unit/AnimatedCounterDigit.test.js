import { shallowToJson } from 'enzyme-to-json'
import { AnimatedCounterDigitBuilder as Builder } from './support/builders'

it('matches snapshot', function () {
  const component = Builder.shallow()
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

it('limits the value to maxValue', function () {
  const component = Builder.shallow({ maxValue: 5 })
  expect(component.find('.rollex-digit-lane-label').length).toBe(7)
})

test('reset() is being called on transition end', function () {
  const component = Builder.mount()
  component.instance().reset = jest.genMockFunction()
  component.instance().forceUpdate() // required for the mock to work
  component.find('.rollex-digit').simulate('transitionEnd')
  expect(component.instance().reset.mock.calls.length).toBe(1)
})
