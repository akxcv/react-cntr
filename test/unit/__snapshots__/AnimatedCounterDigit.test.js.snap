// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="rollex-digit"
  onTransitionEnd={[Function]}
  style={
    Object {
      "MsTransform": "translate3D(0, -9.090909090909092%, 0)",
      "WebkitTransform": "translate3D(0, -9.090909090909092%, 0)",
      "transform": "translate3D(0, -9.090909090909092%, 0)",
      "transitionDuration": "300ms",
      "transitionProperty": "transform",
      "transitionTimingFunction": "ease-in",
      "willChange": "transform",
    }
  }
>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      0
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      1
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      2
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      3
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      4
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      5
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      6
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      7
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      8
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      9
    </span>
  </div>
  <div
    className="rollex-digit-lane-label"
  >
    <span>
      0
    </span>
  </div>
</div>
`;
